### ISSUE TITLE
Curried map

### ISSUE THREAD

**Issue by Ramblurr** [types/npm-ramda#171]
Using the curried map with an Array doesn't work.

**Example**: doesn't compile
```js
function testMap() {
  const data = [
    {id: 1, name: "Alice"},
    {id: 2, name: "Bob"},
  ];
  const nameProp = R.prop("name");
  const nameMap = R.map(nameProp);
  const names: string[] = nameMap(data); // error here
  
  // expected => ["Alice", "Bob"]
}
```
Instead the TS error is:
> TS2322:Type 'Functor<any>' is not assignable to type 'string[]'. Property 'includes' is missing in type 'Functor<any>'.

This is the type def for the one-parameter map:

```js
        // object

        // mixed:
        map<T, U>(fn: (x: T) => U): {
          <M extends Obj<T>>(obj: M): Obj<U>;
          <K extends string>(obj: Obj<T>): Obj<U>;
          (obj: Functor<T>): Functor<U>;
          (list: T[]): U[];
        };
```

**Comment by tycho01** [comment]
Thanks for filing this, I remember bumping into it as well. I'd yet to figure it out though. That said, I'd wonder if this is an order thing (maybe the array version should come above functor)?

**Comment by tycho01** [comment]
Should be fixed with the refactor from #190
